compiler/rustc_parse/src/parser/item.rs

```diff
@@ -2385,7 +2385,22 @@ impl<'a> Parser<'a> {
             };
             let (pat, ty) = if is_name_required || this.is_named_param() {
                 debug!("parse_param_general parse_pat (is_name_required:{})", is_name_required);
-
+                if let token::Ident(symname, _) = this.token.kind
+                    && symname.as_str().eq("Sized")
+                    && this.prev_token.is_lifetime()
+                {
+                    // parse pattern with "'a Sized" we're supposed to give suggestion like
+                    // "'a + Sized"
+                    let mut err = this.struct_span_err(this.prev_token.span, "expected `+` between lifetime and trait");
+                    err.multipart_suggestion(
+                    "missing `+`",
+                     vec![(this.prev_token.span.shrink_to_hi(), " +".to_string()),
+                       (this.token.span.shrink_to_lo(), "".to_string())],
```

Comment:
This doesn't add anything? It's replacing "" with "".

Reply:
If I don't add the second suggestion-part. The emit would look like:
error: expected `+` between lifetime and Sized
```
 --> main.rs:1:20
  |
1 | fn foo<'a>(_: impl 'a Sized) {}
  |                    ^^- help: add `+`: `+`
```
If now is like:
```
error: expected `+` between lifetime and Sized
 --> main.rs:1:20
  |
1 | fn foo<'a>(_: impl 'a Sized) {}
  |                    ^^
  |
help: add `+`
  |
1 | fn foo<'a>(_: impl 'a + Sized) {}
  |                       +
```
Which one do you prefer? I think the current suggestion looks better

Reply:
You should be getting the span between the previous token and the current token, and replacing that with "` + `", so this doesn't matter.

Reply:
fixed by using `span_suggestion_verbose`. Thank you for the suggestion